src/theia/sfm/feature_extractor_and_matcher.cc

```diff
@@ -94,6 +94,64 @@ void ExtractFeatures(
           << " features from image " << image_filepath;
 }
 
+void ExtractFeaturesWithMask(
```

Comment:
Avoid the redundancy is a very good idea.
But, I'm not sure to have well understood your suggestion.
Is it imply to have only one method with an optional parameter for the mask name, as:
`void ExtractFeatures(
    const FeatureExtractorAndMatcher::Options& options,
    const std::string& image_filepath,
    std::vector<Keypoint>* keypoints,
    std::vector<Eigen::VectorXf>* descriptors,
    const std::string& mask_filepath = "")`
and filter points `if (mask_filepath != "")`?